version: RoboCar-V1.0.0

#Serial port param
port_name: /dev/ttyAMA0  #设备端口
baud_rate: 115200       #通信波特率，固定115200

#odom_frame: odom_raw       #里程计坐标系
base_frame: base_link  #底盘坐标系
#publish_odom_transform: False #是否发布odom->base_footprint坐标变换

#Mecanum params
encoder_resolution: 1440  #编码器分辨率
wheel_diameter: 0.0816     #轮胎直径
gear_reduction: 1.0       #电机减速比
wheel_a_mec: 0.095        #麦克纳姆运动模型A参数    
wheel_b_mec: 0.075        #麦克纳姆运动模型B参数
linear_correction_factor: 1.000    #线速度校准系数
angular_correction_factor: 1.000  #角速度校准系数

#Lidar params
lidar_offset_x: 0.0           #激光雷达X轴偏移距离(m)
lidar_offset_y: 0.0           #激光雷达Y轴偏移距离(m)
lidar_offset_z: 0.0           #激光雷达Z轴偏移距离(m)
lidar_offset_roll: 0.0        #激光雷达Roll偏差 (rad)
lidar_offset_pitch: 0.0       #激光雷达Pitch偏差(rad)
lidar_offset_yaw: 3.1415926   #激光雷达Yaw偏差  (rad)
lidar_frame: laser            #激光雷达坐标系名称

#Motor PID params
pid_rate:  20           #OpenCRP PID控制频率  
Kp: 130                 #OpenCRP电机控制PID，P参数
Ki: 0                   #OpenCRP电机控制PID，I参数
Kd: 30                 #OpenCRP电机控制PID, D参数

#Other
control_rate: 20        #底盘控制频率
start_rotation_limit_w: 0.4   #底盘自转最小启动速度
encoder_min: -32768           #编码器最小溢出值
encoder_max: 32767            #编码器最大溢出值